Documentation/compatibility/uri-unicode-bidirectional-characters.md

```diff
@@ -0,0 +1,38 @@
+## Allow Unicode Bidirectional Control Characters in URIs
+
+### Scope
+Minor
+
+### Version Introduced
+4.7.2
+
+### Source Analyzer Status
+NotPlanned
+
+### Change Description
+Unicode specifies several special control characters used to specify the orientation of text.
+In previous versions of the .NET Framework, these characters would be incorrectly stripped
+from all URIs. The characters would be stripped even if they were present in their percent
+encoded form. In order to better follow [RFC 3987](http://tools.ietf.org/html/rfc3987), we now allow these characters in URIs.
+When found un-encoded in a URI they will be percent encoded. When found percent encoded they
+will be left as-is.
+
+- [x] Quirked
+- [ ] Build-time break
+
+### Recommended Action
+The new behavior is enabled by default, but can be disabled by setting the app context switch
+`Switch.System.Uri.DontKeepUnicodeBidiFormattingCharacters` to `true`. When applications target
+earlier versions of the framework but run on 4.7.2, they can enable the fix by instead setting
+the switch to `false`.
```

Comment:
I find this text confusing.  That probably stems from the unfortunate fact that this appcontext switch is negative, i.e. setting to true disables the new behavior.  Yet, by saying "the new behavior is enabled by default"....really means that the app context switch is FALSE by default.  Except, this isn't true when targeting less than .NET 4.7.2...where the default for the switch is TRUE (thus new behavior is OFF).

So, perhaps the text could read something like this:

"The new behavior is enabled by default, except when targeting versions of .NET Framework less than 4.7.2. Setting the app context switch `Switch.System.Uri.DontKeepUnicodeBidiFormattingCharacters` to `true` will disable the new behavior."

cc: @mairaw

Reply:
In response to David's comment, I'd phrase this as:

For applications that target versions of .NET Framework starting with 4.7.2, support for Unicode bidirectional characters is enabled by default. If this change is undesirable, you can disable it by adding the following \[AppContextSwitchOverrides](\~/docs/framework/configure-apps/file-schema/runtime/appcontextswitchoverrides-element.md) switch to the \`\<runtime>\` section of the application configuration file:
\`\`\`xml
\<runtime>
    \<AppContextSwitchOverrides value="Switch.System.Uri.DontKeepUnicodeBidiFormattingCharacters=false" />
\</runtime>
\`\`\`
For applications that target earlier versions of the .NET Framework but are running under versions starting with .NET Framework 4.7.2, support for Unicode bidirectional characters is disabled by default. You can enable it by adding the following \[AppContextSwitchOverrides](~/docs/framework/configure-apps/file-schema/runtime/appcontextswitchoverrides-element.md) switch to the \`\<runtime>\` section of the application configuration file::
\`\`\`xml
\<runtime>
    \<AppContextSwitchOverrides value="Switch.System.Uri.DontKeepUnicodeBidiFormattingCharacters=true" />
\</runtime>
\`\`\`